wherehows-dao/src/main/java/wherehows/dao/table/DictDatasetDao.java

```diff
@@ -30,4 +47,126 @@ public DictDataset findByUrn(String urn) {
   public DictDataset findById(int datasetId) {
     return findBy(DictDataset.class, "id", datasetId);
   }
+
+  /**
+   * Insert or update dict dataset table given information from MetadataChangeEvent
+   * @param identifier DatasetIdentifier
+   * @param auditStamp ChangeAuditStamp
+   * @param property DatasetProperty
+   * @param schema DatasetSchema
+   * @param deployments List<DeploymentDetail>
+   * @param tags List<String>
+   * @param capacities List<Capacity>
+   * @return dataset id
+   * @throws Exception
+   */
+  public DictDataset insertUpdateDataset(DatasetIdentifier identifier, ChangeAuditStamp auditStamp,
+      DatasetProperty property, DatasetSchema schema, List<DeploymentDetail> deployments, List<String> tags,
+      List<Capacity> capacities, PartitionSpecification partitions) throws Exception {
+
+    String urn = toWhDatasetUrn(identifier);
+
+    // find dataset if exist
+    DictDataset dataset = null;
+    try {
+      dataset = findBy(DictDataset.class, "urn", urn);
+    } catch (Exception e) {
+      log.debug("Can't find dataset " + urn, e.toString());
+    }
+    // if not found, create new entity
+    if (dataset == null) {
+      dataset = new DictDataset();
+    }
+
+    // fill in information
+    fillDictDataset(dataset, urn, auditStamp, property, schema, deployments, tags, capacities, partitions);
+
+    // merge into table
+    return (DictDataset) update(dataset);
+  }
+
+  /**
+   * Fill in DictDataset information
+   * @param ds DictDataset
+   * @param urn String
+   * @param auditStamp ChangeAuditStamp
+   * @param property DatasetProperty
+   * @param schema DatasetSchema
+   * @param deployments List<DeploymentDetail>
+   * @param tags List<String>
+   * @param capacities List<Capacity>
+   * @param partitions PartitionSpecification
+   * @throws IOException
+   */
+  public void fillDictDataset(DictDataset ds, String urn, ChangeAuditStamp auditStamp, DatasetProperty property,
+      DatasetSchema schema, List<DeploymentDetail> deployments, List<String> tags, List<Capacity> capacities,
+      PartitionSpecification partitions) throws IOException {
+
+    ObjectMapper mapper = new ObjectMapper();
```

Comment:
Probably we can initialize an ObjectMapper() in baseDao, it will be use in many places.